### ISSUE TITLE
Missing ['emotion'] column in test.csv?

### ISSUE THREAD

**Issue by weimun19** [omarsayed7/Deep-Emotion#6]
I have tried to run the code with the train and test data from Kaggle, but it went wrong when running the main.py for training the model, and I found the problem is that the plain_dataset() in data_loaders.py trying to retrieve the ['emotion'] column in val.csv which doesn't exist. How can I solve this problem?

**Comment by Cip0** [comment]
Downloading the fer2013.csv and splitting it into train.csv and test.csv (80/20 proportion) worked for me

**Comment by wengwenwen96** [comment]
When I train this network, the loss function and accuracy remain unchanged.

Epoch: 1        Training Loss: 0.01437178       Validation Loss 0.01457405      Training Acuuarcy 24.926%       Validation Acuuarcy 24.937%
Epoch: 2        Training Loss: 0.01419203       Validation Loss 0.01462503      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 3        Training Loss: 0.01419188       Validation Loss 0.01466722      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 4        Training Loss: 0.01419332       Validation Loss 0.01468434      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 5        Training Loss: 0.01419407       Validation Loss 0.01456469      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 6        Training Loss: 0.01419189       Validation Loss 0.01469155      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 7        Training Loss: 0.01419034       Validation Loss 0.01467826      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 8        Training Loss: 0.01419431       Validation Loss 0.01466055      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 9        Training Loss: 0.01418729       Validation Loss 0.01462144      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 10       Training Loss: 0.01418980       Validation Loss 0.01466851      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 11       Training Loss: 0.01419380       Validation Loss 0.01461501      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 12       Training Loss: 0.01418736       Validation Loss 0.01464282      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 13       Training Loss: 0.01418842       Validation Loss 0.01461904      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 14       Training Loss: 0.01418909       Validation Loss 0.01475997      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 15       Training Loss: 0.01418691       Validation Loss 0.01454415      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 16       Training Loss: 0.01419212       Validation Loss 0.01459282      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 17       Training Loss: 0.01419064       Validation Loss 0.01479096      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 18       Training Loss: 0.01418800       Validation Loss 0.01460306      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 19       Training Loss: 0.01418691       Validation Loss 0.01465107      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 20       Training Loss: 0.01418918       Validation Loss 0.01462860      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 21       Training Loss: 0.01419190       Validation Loss 0.01458804      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 22       Training Loss: 0.01418829       Validation Loss 0.01460765      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 23       Training Loss: 0.01418722       Validation Loss 0.01455541      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 24       Training Loss: 0.01418603       Validation Loss 0.01469193      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 25       Training Loss: 0.01418951       Validation Loss 0.01463459      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 26       Training Loss: 0.01418810       Validation Loss 0.01476145      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 27       Training Loss: 0.01418882       Validation Loss 0.01468886      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 28       Training Loss: 0.01419031       Validation Loss 0.01463278      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%
Epoch: 29       Training Loss: 0.01418738       Validation Loss 0.01465452      Training Acuuarcy 25.131%       Validation Acuuarcy 24.937%

**Comment by omarsayed7** [comment]
> When I train this network, the loss function and accuracy remain unchanged.
> 
> Epoch: 1 Training Loss: 0.01437178 Validation Loss 0.01457405 Training Acuuarcy 24.926% Validation Acuuarcy 24.937%
> Epoch: 2 Training Loss: 0.01419203 Validation Loss 0.01462503 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 3 Training Loss: 0.01419188 Validation Loss 0.01466722 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 4 Training Loss: 0.01419332 Validation Loss 0.01468434 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 5 Training Loss: 0.01419407 Validation Loss 0.01456469 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 6 Training Loss: 0.01419189 Validation Loss 0.01469155 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 7 Training Loss: 0.01419034 Validation Loss 0.01467826 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 8 Training Loss: 0.01419431 Validation Loss 0.01466055 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 9 Training Loss: 0.01418729 Validation Loss 0.01462144 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 10 Training Loss: 0.01418980 Validation Loss 0.01466851 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 11 Training Loss: 0.01419380 Validation Loss 0.01461501 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 12 Training Loss: 0.01418736 Validation Loss 0.01464282 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 13 Training Loss: 0.01418842 Validation Loss 0.01461904 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 14 Training Loss: 0.01418909 Validation Loss 0.01475997 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 15 Training Loss: 0.01418691 Validation Loss 0.01454415 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 16 Training Loss: 0.01419212 Validation Loss 0.01459282 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 17 Training Loss: 0.01419064 Validation Loss 0.01479096 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 18 Training Loss: 0.01418800 Validation Loss 0.01460306 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 19 Training Loss: 0.01418691 Validation Loss 0.01465107 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 20 Training Loss: 0.01418918 Validation Loss 0.01462860 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 21 Training Loss: 0.01419190 Validation Loss 0.01458804 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 22 Training Loss: 0.01418829 Validation Loss 0.01460765 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 23 Training Loss: 0.01418722 Validation Loss 0.01455541 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 24 Training Loss: 0.01418603 Validation Loss 0.01469193 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 25 Training Loss: 0.01418951 Validation Loss 0.01463459 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 26 Training Loss: 0.01418810 Validation Loss 0.01476145 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 27 Training Loss: 0.01418882 Validation Loss 0.01468886 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 28 Training Loss: 0.01419031 Validation Loss 0.01463278 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%
> Epoch: 29 Training Loss: 0.01418738 Validation Loss 0.01465452 Training Acuuarcy 25.131% Validation Acuuarcy 24.937%

try to follow the readme and also it might help to read this comment https://github.com/omarsayed7/Deep-Emotion/issues/6#issuecomment-734877485
